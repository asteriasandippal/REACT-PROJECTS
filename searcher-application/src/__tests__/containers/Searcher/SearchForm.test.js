import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import {stub} from 'sinon';

import SearForm from '../../../containers/Searcher/SearchForm';

test('Search Form render correctly', () => {
    const wrapper = renderer.create(<SearForm onSearch={() => {}}/>);
    const tree  = wrapper.toJSON();

    expect(tree).toMatchSnapshot();
});

test('SearchForm calls onSearch with the contents of the input when the form is submitted', () => {
    
    const onSearchStub = stub();
    const wrapper = mount(
        <SearForm onSearch={onSearchStub}/>
    );

    wrapper.find('input').simulate('change', {target: {value: 'Test'}});
    wrapper.find('button').simulate('submit');

    expect(onSearchStub.callCount).toBe(1);
    expect(onSearchStub.calledWith('Test')).toBe(true);
});