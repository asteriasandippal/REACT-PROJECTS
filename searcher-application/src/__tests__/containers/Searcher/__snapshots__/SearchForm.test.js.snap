// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form render correctly 1`] = `
<div>
  <form
    className="pt-4 pb-4"
    onSubmit={[Function]}
  >
    <div
      className="input-group mb-3"
    >
      <input
        aria-label="Search phots..."
        className="form-control form-control-lg"
        onChange={[Function]}
        placeholder="Search phots..."
        type="text"
      />
      <div
        className="input-group-append"
      >
        <button
          className="btn btn-primary btn-lg"
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </form>
</div>
`;
